src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -65,6 +65,7 @@ public MethodKind MethodKind
             public bool IsExtensionMethod
             {
                 get { return (_flags & IsExtensionMethodBit) != 0; }
+                set { _flags = value ? (_flags | IsExtensionMethodBit) : (_flags & ~IsExtensionMethodBit); }
```

Comment:
From what I can see. This is just storing information about `SourceMethodSymbol` in a compact way. Things that can be calculated at once during the symbol constructor like `MethodKind` don't have setters. Things that require binding for its value like `ReturnsVoid` have both setters and getters. They are initialized with dummy values then set later during `MethodChecks` to its correct value.
Please see `Flags.ReturnsVoid` and `SetReturnsVoid()`.

---
In reply to: [122884575](https://github.com/dotnet/roslyn/pull/19703#discussion_r122884575) [](ancestors = 122884575)